app/directives/proposalList/proposalList.directive.js

```diff
@@ -247,7 +247,20 @@ angular.module('dashboard')
             self.add = function (data) {
                 $log.debug("dbProposalList: add: " + JSON.stringify(data));
                 if (angular.isObject(data) && angular.isObject(data.proposal)) {
-                    self.proposals.splice(0, 0, data.proposal);
+
+                    var notExists = true;
```

Comment:
"not" naming is always a bit confusing, could it be instead "exists"?
Compare for example potential code ìf (!notExists)` or `if (exists)`